File: app/src/main/java/com/test/xyz/demo/presentation/weather/di/WeatherFragmentComponent.java
package com.test.xyz.demo.presentation.weather.di;

import com.test.xyz.demo.presentation.common.di.scope.ActivityScope;
import com.test.xyz.demo.presentation.weather.WeatherFragment;

import dagger.Subcomponent;

@ActivityScope
@Subcomponent(
        modules = {WeatherFragmentModule.class}
)
public interface WeatherFragmentComponent {
    void inject(WeatherFragment mainFragment);
}

File: app/src/main/kotlin/com/test/xyz/demo/presentation/weather/di/WeatherFragmentComponent.kt
package com.test.xyz.demo.presentation.weather.di

import com.test.xyz.demo.presentation.common.di.scope.ActivityScope
import com.test.xyz.demo.presentation.weather.WeatherFragment

import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = arrayOf(WeatherFragmentModule::class))
interface WeatherFragmentComponent {
    fun inject(mainFragment: WeatherFragment)
}

